crypto/ct/ct_sct.c

```diff
@@ -372,10 +377,28 @@ int SCT_validate(SCT *sct, const CT_POLICY_EVAL_CTX *ctx)
             goto err;
     }
 
+    /*
+     * XXX: Potential for optimization.  This repeats some idempotent heavy
+     * lifting on the certificate for each candidate SCT, and appears to not
+     * use any information in the SCT itself, only the certificate is
+     * processed.  So it may make more sense to to do this just once, perhaps
+     * associated with the shared (by all SCTs) policy eval ctx.
+     *
+     * XXX: Failure here is global (SCT independent) and represents either an issue
+     * with the certificate (e.g. duplicate extensions) or an out of memory
+     * condition.  When the certificate is incompatible with CT, we really
```

Comment:
I had trouble with the "When the certificate..." sentence. Does it mean

"When the certificate is incompatible, we mark the SCTs invalid, rather than report a fatal error. That way, callbacks that want to do "soft" SCT processing will not abort handshakes with false positive certificate errors."

I.e., replace "we should mark" with "we mark" - it's what we do, after your change. And replace "false positive internal errors" with "false positive certificate errors" because the internal errors are not false positive from a "soft" CT callback's perspective - you'd normally want to abort on a malloc failure.